assets/js/googlesitekit/datastore/user/surveys.test.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * `core/user` data store: Surveys tests.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	createTestRegistry,
+} from '../../../../../tests/js/utils';
+import {
+	act,
+} from '../../../../../tests/js/test-utils';
+import { STORE_NAME } from './constants';
+
+describe( 'core/user surveys', () => {
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+	} );
+
+	const survey = {
+		survey_payload: 'foo',
+		session: 'bar',
+	};
+
+	describe( 'actions', () => {
+		describe( 'triggerSurvey', () => {
+			it( 'throws an error when parameters are missing or incorrect', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey();
+				} ).toThrow( 'triggerID is mandatory and must be a string' );
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey( 'ABCD', false );
+				} ).toThrow( 'options must be an object' );
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey( 'ABCD', { ttl: 'a' } );
+				} ).toThrow( 'options.ttl must be a number' );
+			} );
+
+			it( 'does not throw an error when parameters are correct', () => {
+				fetchMock.post(
```

Comment:
I believe this should be `freezeFetch` because you're not doing anything with the response?

Reply:
Updated in https://github.com/google/site-kit-wp/pull/3431/commits/a9782e9929559ca0f0020d853f6eacc5758d12aa